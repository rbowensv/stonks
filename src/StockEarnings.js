import React from 'react'
import doge from './tothemoon.gif'
import sell from './homersell.gif'

export const StockEarnings = (props) => {
    const { earnings } = props;

    const isProfitable = earnings ? earnings.thisQuarter > earnings.lastQuarter : null;

    const proffitClass = isProfitable ? 'stock-earnings--green' : 'stock-earnings--red';

    return <div className='stock-earnings'>Earnings:
                {earnings ? <div className='stock-earnings--row'>
                    <div className={proffitClass}>
                        {isProfitable ? 
                            <img className='stock-earnings--meme' src={doge} alt='doge in rocket, text to the moon flashing rainbow'/> : 
                            <img className='stock-earnings--meme' src={sell} alt='homer simpson shout to phone Sell!'/>
                        }
                    </div> 
                    <div className='stock-earnings--section'>
                        <div>Reported Earnings: {earnings['thisQuarter']}</div>
                        <div>{isProfitable ? '+ ' : ''}{earnings.thisQuarter/earnings.lastQuarter*100 - 100}%</div>
                    </div> 
                </div> : <div>
                    Something went wrong
                </div>}
            </div>
}