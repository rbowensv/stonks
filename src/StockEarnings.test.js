import renderer from 'react-test-renderer'
import { StockEarnings } from './StockEarnings'
import doge from './tothemoon.gif'
import sell from './homersell.gif'

const mockEarningsGood = {thisQuarter: "0.5762", lastQuarter: "0.2809"}
const mockEarnignsBad = {thisQuarter: "0.2809", lastQuarter: "0.5762"}

test('renders as expected', () => {
    const component = renderer.create(<StockEarnings earnings={mockEarningsGood}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// test('doge when good', () => {
//     const component = renderer.create(<StockEarnings earnings={mockEarningsGood}/>);
//     expect(component.root).toContainHTML("<img className='stock-earnings--meme' src={doge} alt='doge in rocket, text to the moon flashing rainbow'/>")
// });

// test('homer when bad', () => {
//     const component = renderer.create(<StockEarnings earnings={mockEarnignsBad}/>);
//     expect(component).toContainHTML(<img className='stock-earnings--meme' src={sell} alt='homer simpson shout to phone Sell!'/>)
// })