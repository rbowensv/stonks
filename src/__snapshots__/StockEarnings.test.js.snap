// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="stock-earnings"
>
  Earnings:
  <div
    className="stock-earnings--row"
  >
    <div
      className="stock-earnings--green"
    >
      <img
        alt="doge in rocket, text to the moon flashing rainbow"
        className="stock-earnings--meme"
        src="tothemoon.gif"
      />
    </div>
    <div
      className="stock-earnings--section"
    >
      <div>
        Reported Earnings: 
        0.5762
      </div>
      <div>
        + 
        105.12637949448208
        %
      </div>
    </div>
  </div>
</div>
`;
